[2023-12-21T16:55:03.092+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings.impute_DeafOrHardOfHearing_with_mode scheduled__2023-12-01T00:00:00+00:00 [queued]>
[2023-12-21T16:55:03.097+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings.impute_DeafOrHardOfHearing_with_mode scheduled__2023-12-01T00:00:00+00:00 [queued]>
[2023-12-21T16:55:03.097+0000] {taskinstance.py:1361} INFO - Starting attempt 2 of 6
[2023-12-21T16:55:03.104+0000] {taskinstance.py:1382} INFO - Executing <Task(PostgresOperator): impute_DeafOrHardOfHearing_with_mode> on 2023-12-01 00:00:00+00:00
[2023-12-21T16:55:03.108+0000] {standard_task_runner.py:57} INFO - Started process 1209 to run task
[2023-12-21T16:55:03.111+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings', 'impute_DeafOrHardOfHearing_with_mode', 'scheduled__2023-12-01T00:00:00+00:00', '--job-id', '69', '--raw', '--subdir', 'DAGS_FOLDER/Indicators_of_Heart_Disease_2022.py', '--cfg-path', '/tmp/tmp_be9miw3']
[2023-12-21T16:55:03.113+0000] {standard_task_runner.py:85} INFO - Job 69: Subtask impute_DeafOrHardOfHearing_with_mode
[2023-12-21T16:55:03.147+0000] {task_command.py:416} INFO - Running <TaskInstance: Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings.impute_DeafOrHardOfHearing_with_mode scheduled__2023-12-01T00:00:00+00:00 [running]> on host 8e01071a5eb3
[2023-12-21T16:55:03.195+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='dunnbc22' AIRFLOW_CTX_DAG_ID='Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings' AIRFLOW_CTX_TASK_ID='impute_DeafOrHardOfHearing_with_mode' AIRFLOW_CTX_EXECUTION_DATE='2023-12-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-12-01T00:00:00+00:00'
[2023-12-21T16:55:03.196+0000] {sql.py:274} INFO - Executing: -- params: table_name, column_name

UPDATE indicators_of_heart_disease_2022_pipe
SET DeafOrHardOfHearing = (
    SELECT DeafOrHardOfHearing 
    FROM indicators_of_heart_disease_2022_pipe
    WHERE DeafOrHardOfHearing IS NOT NULL
    GROUP BY DeafOrHardOfHearing
    ORDER BY COUNT(*) DESC
    LIMIT 1
)
WHERE DeafOrHardOfHearing IS NULL;
[2023-12-21T16:55:03.203+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2023-12-21T16:55:03.209+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2023-12-21T16:55:03.216+0000] {sql.py:418} INFO - Running statement: -- params: table_name, column_name

UPDATE indicators_of_heart_disease_2022_pipe
SET DeafOrHardOfHearing = (
    SELECT DeafOrHardOfHearing 
    FROM indicators_of_heart_disease_2022_pipe
    WHERE DeafOrHardOfHearing IS NOT NULL
    GROUP BY DeafOrHardOfHearing
    ORDER BY COUNT(*) DESC
    LIMIT 1
)
WHERE DeafOrHardOfHearing IS NULL;, parameters: None
[2023-12-21T16:55:05.506+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/operators/sql.py", line 280, in execute
    output = hook.run(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 423, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.DeadlockDetected: deadlock detected
DETAIL:  Process 74277 waits for ShareLock on transaction 1141772; blocked by process 74275.
Process 74275 waits for ShareLock on transaction 1141764; blocked by process 74266.
Process 74266 waits for ShareLock on transaction 1141767; blocked by process 74268.
Process 74268 waits for ShareLock on transaction 1141771; blocked by process 74277.
HINT:  See server log for query details.
CONTEXT:  while updating tuple (19758,1) in relation "indicators_of_heart_disease_2022_pipe"

[2023-12-21T16:55:05.520+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings, task_id=impute_DeafOrHardOfHearing_with_mode, execution_date=20231201T000000, start_date=20231221T165503, end_date=20231221T165505
[2023-12-21T16:55:05.536+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 69 for task impute_DeafOrHardOfHearing_with_mode (deadlock detected
DETAIL:  Process 74277 waits for ShareLock on transaction 1141772; blocked by process 74275.
Process 74275 waits for ShareLock on transaction 1141764; blocked by process 74266.
Process 74266 waits for ShareLock on transaction 1141767; blocked by process 74268.
Process 74268 waits for ShareLock on transaction 1141771; blocked by process 74277.
HINT:  See server log for query details.
CONTEXT:  while updating tuple (19758,1) in relation "indicators_of_heart_disease_2022_pipe"
; 1209)
[2023-12-21T16:55:05.542+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-12-21T16:55:05.600+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
